Composer/packages/ui-plugins/select-skill-dialog/src/BeginSkillDialogField.tsx

```diff
@@ -6,38 +6,88 @@ import { FieldProps, JSONSchema7, useShellApi } from '@bfc/extension';
 import { Link } from 'office-ui-fabric-react/lib/Link';
 import { ObjectField, SchemaField } from '@bfc/adaptive-form';
 import formatMessage from 'format-message';
+import { Skill } from '@bfc/shared';
 
 import { SkillEndpointField } from './SkillEndpointField';
 
+const referBySettings = (skillName: string, property: string) => {
+  return `=settings.skill['${skillName}'].${property}`;
+};
+
+const handleBackwardCompat = (skills: Skill[], value): { name: string; referredUpdates: any } | undefined => {
+  const { skillEndpoint } = value;
+  const foundSkill = skills.find(({ manifestUrl }) => manifestUrl === value.id);
+  let updates: any = {};
+  if (foundSkill) {
+    updates = {
+      id: referBySettings(foundSkill.name, 'manifestUrl'),
+    };
+    const matchedEndpoint: any = foundSkill.endpoints.find(({ endpointUrl }) => endpointUrl === skillEndpoint);
+    if (matchedEndpoint) {
+      updates.skillEndpoint = referBySettings(foundSkill.name, 'endpointUrl');
+      updates.skillAppId = referBySettings(foundSkill.name, 'msAppId');
+    }
+
+    return {
+      name: foundSkill?.name,
+      referredUpdates: updates,
+    };
+  }
+};
+
 export const BeginSkillDialogField: React.FC<FieldProps> = (props) => {
   const { depth, id, schema, uiOptions, value, onChange, definitions } = props;
   const { projectId, shellApi, skills = [] } = useShellApi();
-  const { displayManifestModal } = shellApi;
+  const { displayManifestModal, skillsInSettings } = shellApi;
 
-  const manifest = useMemo(() => skills.find(({ manifestUrl }) => manifestUrl === value.id), [skills, value.id]);
-  const endpointOptions = useMemo(() => (manifest?.endpoints || []).map(({ name }) => name), [manifest]);
+  const manifest: Skill | undefined = useMemo(() => {
+    const matchedSkill = skills.find(({ manifestUrl }) => {
+      return manifestUrl === skillsInSettings.get(value.id);
+    });
+    // Handle backward compatibility
+    if (!matchedSkill) {
+      const { id, skillEndpoint, skillAppId, ...rest } = value;
+      const result = handleBackwardCompat(skills, value);
+      if (result) {
+        onChange({ ...rest, ...result.referredUpdates });
+      }
+    }
+    return matchedSkill;
+  }, [skills, value.id]);
+
+  const endpointOptions = useMemo(() => {
+    return (manifest?.endpoints || []).map(({ name }) => name);
+  }, [manifest]);
 
-  const handleIdChange = ({ key }) => {
+  const handleIdChange = ({ key, text }) => {
     if (!manifest || key !== manifest.manifestUrl) {
       const { skillEndpoint, skillAppId, ...rest } = value;
-      onChange({ ...rest, id: key });
+      onChange({ ...rest, id: referBySettings(text, 'manifestUrl') });
```

Comment:
Previously, we were only using the `id` field to correctly populate the 'Skill endpoint' dropdown; however, since the 'Skill endpoint' is mostly static, I think we can move away from setting the `id` field and populate the dropdown options based on the `skillName` in `=settings.skill['${skillName}'].skillEndpoint`.

Reply:
`Id` the non mandatory field is no longer set in the schema